compiler/src/main/scala/org/scalajs/core/compiler/JSGlobalAddons.scala

```diff
@@ -45,6 +45,50 @@ trait JSGlobalAddons extends JSDefinitions
     case object Static extends ExportDestination
   }
 
+  /** Extracts the super type of a `Template`, with type parameters reinvented
+   *  so that the type is well-formed outside of the `Template`, i.e., at the
+   *  same level where the corresponding `ImplDef` is defined.
+   */
+  def extractSuperTpeFromImpl(impl: Template): Type =
+    reinventTypeParams(impl.parents.head.tpe)
+
+  /** Reinvents all the type parameters of a `TypeRef`.
+   *
+   *  This is done by existentially quantifying over all type parameters of
+   *  the class type referenced by the `TypeRef`.
+   *
+   *  As a simple example, given the definition
+   *  {{{
+   *  class C[A, B <: AnyVal]
+   *  }}}
+   *  this transforms
+   *  {{{
+   *  path.C[A, Int]
+   *  }}}
+   *  into
+   *  {{{
+   *  path.C[_, _ <: AnyVal]
+   *  }}}
+   *
+   *  As a complex example, given the definition
+   *  {{{
+   *  class D[A, B <: List[Seq[A]]]
+   *  }}}
+   *  this method transforms
+   *  {{{
+   *  path.D[?0, ?1] forSome { type ?0; type ?1 <: List[Seq[?0]] }
+   *  }}}
+   */
+  def reinventTypeParams(tp: Type): Type = {
```

Comment:
Is this used outside of `extractSuperTpeFromImpl`? If no, make private?

Reply:
It used to be. Not anymore, though, so I'll make it private :)